numba_cuda/numba/cuda/tests/support.py

```diff
@@ -817,3 +821,28 @@ def check_warnings(self, messages, category=RuntimeWarning):
                     self.assertEqual(w.category, category)
                     found += 1
         self.assertEqual(found, len(messages))
+
+
+class SerialMixin(object):
+    """Mixin to mark test for serial execution."""
+
+    _numba_parallel_test_ = False
+
+
+class IRPreservingTestPipeline(CompilerBase):
+    """Same as the standard pipeline, but preserves the func_ir into the
+    metadata store after legalisation, useful for testing IR changes"""
+
+    def define_pipelines(self):
+        pipeline = DefaultPassBuilder.define_nopython_pipeline(
+            self.state, "ir_preserving_custom_pipe"
+        )
+        # mangle the default pipeline and inject DCE and IR preservation ahead
+        # of legalisation
+
+        # TODO: add a way to not do this! un-finalizing is not a good idea
+        pipeline._finalized = False
+        pipeline.add_pass_after(PreserveIR, IRLegalization)
+
+        pipeline.finalize()
+        return [pipeline]
```

Comment:
I think this is no longer used in this PR (and we would not be able to use it in Numba-CUDA tests anyway, because we don't support the `pipeline_class` kwarg):
```suggestion
```